File: step2/lect1/selectionsort/SelectionSort.java
class SelectionSort {
    public static void main(String[] args) {
        int[] arr = new int[] { 1, 4, 6, 5, 7, 3, 6 };

        (new SelectionSort()).selectionSort(arr);
    }

    void selectionSort(int[] arr) {
        int n = arr.length;

        for (int i = 0; i < n - 1; i++) {
            int min = arr[i];
            int k = i;
            for (int j = i; j <= n - 1; j++) {
                if (arr[j] <= min) {
                    min = arr[j];
                    k = j;
                }
            }

            arr[k] = arr[i];
            arr[i] = min;

            print(arr);
        }
    }

    void print(int[] arr) {
        for (int i : arr) {
            System.out.print(i + " ");
        }

        System.err.println();
    }
}

File: step2/lect1/selectionsort/selectionsort.go
package main

import "fmt"

func main() {
	arr := [7]int{3, 4, 2, 7, 1, 9, 6}

	for i := range arr {
		min := arr[i]

		k := i

		for j := i + 1; j < len(arr); j++ {
			if arr[j] <= min {
				min = arr[j]
				k = j
			}
		}

		arr[k] = arr[i]
		arr[i] = min

		fmt.Println(arr)

	}
}
